zeebe/dynamic-config/src/test/java/io/camunda/zeebe/dynamic/config/changes/DeleteHistoryApplierTest.java

```diff
@@ -0,0 +1,93 @@
+import java.util.Map;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class DeleteHistoryApplierTest {
+
+  @Test
+  void shouldSuccessDeleteHistoryOnInitIfNoPartitionExists() {
+    // given
+    final MemberId memberId = MemberId.from("1");
+    final ClusterChangeExecutor clusterChangeExecutor =
+        new ClusterChangeExecutor.NoopClusterChangeExecutor();
+    final var deleteHistoryApplier = new DeleteHistoryApplier(memberId, clusterChangeExecutor);
+
+    final ClusterConfiguration clusterConfigurationWithMember =
+        ClusterConfiguration.init().addMember(memberId, MemberState.initializeAsActive(Map.of()));
+
+    // when
+    final var result = deleteHistoryApplier.init(clusterConfigurationWithMember);
+
+    // then
+    EitherAssert.assertThat(result).isRight();
+  }
+
+  @Test
+  void shouldFailDeleteHistoryOnInitIfPartitionsExist() {
+    // given
+    final MemberId memberId = MemberId.from("1");
+    final int partitionId = 2;
+    final ClusterChangeExecutor clusterChangeExecutor =
+        new ClusterChangeExecutor.NoopClusterChangeExecutor();
+    final var deleteHistoryApplier = new DeleteHistoryApplier(memberId, clusterChangeExecutor);
+
+    final ClusterConfiguration clusterConfigWithPartition =
+        ClusterConfiguration.init()
+            .addMember(memberId, MemberState.initializeAsActive(Map.of()))
+            .updateMember(
+                memberId,
+                m ->
+                    m.addPartition(
+                        partitionId,
+                        PartitionState.active(
+                            1, new DynamicPartitionConfig(new ExportersConfig(Map.of())))));
+    // when
+    final var result = deleteHistoryApplier.init(clusterConfigWithPartition);
+
+    // then
+    EitherAssert.assertThat(result).isLeft();
+    Assertions.assertThat(result.getLeft())
+        .isInstanceOf(IllegalStateException.class)
+        .hasMessageContaining("partitions still exist");
+  }
+
+  @Test
+  void shouldPreserveSameConfigurationAfterApply() {
+    // given
+    final MemberId member1Id = MemberId.from("1");
+    final MemberId member2Id = MemberId.from("2");
+    final ClusterChangeExecutor clusterChangeExecutor =
+        new ClusterChangeExecutor.NoopClusterChangeExecutor();
+
+    final ClusterConfiguration initialConfiguration =
+        ClusterConfiguration.init()
+            .addMember(member1Id, MemberState.initializeAsActive(Map.of()))
+            .addMember(member2Id, MemberState.initializeAsActive(Map.of()));
+    final var deleteHistoryApplier = new DeleteHistoryApplier(member1Id, clusterChangeExecutor);
+    final var initializedConfiguration =
+        deleteHistoryApplier.init(initialConfiguration).get().apply(initialConfiguration);
+    final var updater = deleteHistoryApplier.apply().join();
+    final ClusterConfiguration updatedConfiguration = updater.apply(initializedConfiguration);
+
+    // when
```

Comment:
:upside_down_face: I assume this is `then`, and the applies above where the `when` ;)